tests/unit_tests/test_dynamic_prompt.py

```diff
@@ -0,0 +1,93 @@
+"""Test functionality related to dynamic prompts."""
+from langchain.prompt import DynamicPrompt, Prompt
+
+
+def test_dynamic_prompt_valid() -> None:
+    """Test dynamic prompt can be successfully constructed from examples."""
+    template = """Test Prompt:
+
+Question: who are you?
+Answer: foo
+
+Question: what are you?
+Answer: bar
+
+Question: {question}
+Answer:"""
+    input_variables = ["question"]
+    example_separator = "\n\n"
+    prefix = """Test Prompt:"""
+    suffix = """Question: {question}\nAnswer:"""
+    examples = [
+        """Question: who are you?\nAnswer: foo""",
+        """Question: what are you?\nAnswer: bar""",
+    ]
+    dynamic_prompt = DynamicPrompt(
+        examples=examples,
+        suffix=suffix,
+        input_variables=input_variables,
+        example_separator=example_separator,
+        prefix=prefix,
+    )
+    prompt_from_template = Prompt(input_variables=input_variables, template=template)
+    assert dynamic_prompt.format(question="foo?") == prompt_from_template.format(
```

Comment:
uber nit: can we do the evaluation of dynamic_prompt and prompt_from_template on their own lines to avoid this uglier/less readable hanging indent? and to make the assert line not contain any logic besides the assert